components/eventing-controller/pkg/handlers/nats_test.go

```diff
@@ -87,29 +89,93 @@ func TestConvertMsgToCE(t *testing.T) {
 	}
 }
 
-func NewNatsMessagePayload(data, id, source, eventTime, eventType string) string {
-	jsonCE := fmt.Sprintf("{\"data\":\"%s\",\"datacontenttype\":\"application/json\",\"id\":\"%s\",\"source\":\"%s\",\"specversion\":\"1.0\",\"time\":\"%s\",\"type\":\"%s\"}", data, id, source, eventTime, eventType)
-	return jsonCE
-}
+func TestSubscription(t *testing.T) {
+	natsPort := 5222
+	subscriberPort := 8080
+	subscriberReceiveURL := fmt.Sprintf("http://127.0.0.1:%d/subscribe", subscriberPort)
+	subscriberCheckURL := fmt.Sprintf("http://127.0.0.1:%d/check", subscriberPort)
 
-func NewCloudEvent(data, id, source, eventTime, eventType string, t *testing.T) cev2event.Event {
-	timeInRFC3339, err := time.Parse(time.RFC3339, eventTime)
+	// Start Nats server
+	natsServer := eventingtesting.RunNatsServerOnPort(natsPort)
+	defer natsServer.Shutdown()
+
+	natsURL := natsServer.ClientURL()
+	natsClient := Nats{
+		Subscriptions: make(map[string]*nats.Subscription),
+		Config: env.NatsConfig{
+			Url:           natsURL,
+			MaxReconnects: 2,
+			ReconnectWait: time.Second,
+		},
+		Log: ctrl.Log.WithName("reconciler").WithName("Subscription"),
+	}
+	err := natsClient.Initialize()
 	if err != nil {
-		t.Fatalf("failed to parse time: %v", err)
+		t.Fatalf("failed to connect to Nats Server: %v", err)
 	}
-	dataContentType := "application/json"
-	return cev2event.Event{
-		Context: &cev2event.EventContextV1{
-			Type: eventType,
-			Source: types.URIRef{
-				URL: url.URL{
-					Path: source,
-				},
-			},
-			ID:              id,
-			DataContentType: &dataContentType,
-			Time:            &types.Timestamp{Time: timeInRFC3339},
-		},
-		DataEncoded: []byte(data),
+
+	// Create a new subscriber
+	subscriber := eventingtesting.NewSubscriber(fmt.Sprintf(":%d", subscriberPort), t)
+	subscriber.Start()
+
+	// Shutting down subscriber
+	defer subscriber.Shutdown()
+
+	// Check subscriber is running or not by checking the store
+	err = subscriber.CheckEvent("", subscriberCheckURL)
+	if err != nil {
+		t.Errorf("subscriber did not receive the event: %v", err)
```

Comment:
The returned error can be also `failed to check the event after retries` which is maybe another reason other than the store is empty

Reply:
True, but we expect the store to be empty at this point. Hence checking for empty value of store will make sure 2 things:
- Subscriber server is working
- And store is clean hence it is good to proceed with the test